### PR TITLE
WE-470

### LINKED ISSUE #168: User story:  Rig Object - API
As a Developer, I want the WE API to support the RIG object so that I can use the WE API to perform operations on this object

### Acceptance criteria ###
- [ ] The WE API is extended to support the RIG object with create, read, update, delete

Also, here are a few points that need to be addressed:
- [ ] Constraint 1
- [ ] Constraint 2

### Resources ###
-  Resource A
-  Resource B

### PR DESCRIPTION
**Pull request by steinsiv**
## Fixes
This pull request fixes # 168 and # 170

## Description
`Modify` and `Delete` for `Rig` object

* New feature (non-breaking change which adds functionality)
* Enhancement of existing functionality

## Impacted Areas in Application
Full stack

*Communication*
* [x] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx [hunk 1]
```diff
@@ -1153,6 +1173,11 @@ export interface SelectRigGroupAction extends Action {
   payload: { well: Well; wellbore: Wellbore; rigGroup: any };
 }
 
+// export interface SelectRigAction extends Action {
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx]: residual comment

File: Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx [hunk 2]
```diff
@@ -1219,11 +1245,11 @@ export type NavigationAction =
   | SelectLogObjectAction
   | SelectWellAction
   | SelectWellboreAction
+  // | SelectRigAction
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx]: residual comment

File: Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx (hunk 1 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx]: Updated

File: Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx (hunk 2 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Frontend/contexts/navigationStateReducer.tsx]: Removed

**Review by janburak — APPROVED** [review]
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**